### DISCUSSION TITLE
[A/L] (2021/22) Relazioni - Esercizio 7

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/metodi-matematici-per-l-informatica#279] [Esercizi A-L]
![image](https://user-images.githubusercontent.com/69389461/152318902-bc8ea185-2f37-4586-b4a2-6e3efc156bdc.png)

**Comment by LuginiAndrea** [comment]
$R$ non è ordine parziale.
Prendiamo $f\left(x\right) = \frac{x}{2}$, e $g\left(x\right) = x - 2$
Prendiamo $S = \{4\}$ (non abbiamo limitazioni sul dominio, quindi possiamo prenderlo come ci pare).
Notiamo che vale $fRg$ in quanto $f\left(4\right) = 2 = g\left(4\right)$ e anche $gRf$ per lo stesso motivo, ma $f \neq g$

**Comment by Exyss** [comment]
@LuginiAndrea la dimostrazione risulta errata poiché una funzione è **anche** una relazione. Difatti, ad esempio, si ha che:
$$f : \mathbb{N} \to \mathbb{R} : x \mapsto\frac{x}{2}  \implies f :  \left \\{(0,0), (1, \frac{1}{2}), (2, 1),  (3, \frac{3}{2}), \ldots \right \\}$$

Dunque, lavorando su esempi numerici, per dimostrare che $f \neq g$ è necessario mostrare che $\exists (a,b) \in f \mid (a,b) \notin g$
(o viceversa)

In particolare, quindi, considerando $S = \\{4\\}, f(x) = \frac{x}{2}$ e $g(x) = x-2$ si avrebbe che $f : \left \\{ (4,2) \right \\}$ e $g : \left \\{ (4,2) \right \\}$, dunque $f = g$

---------

Per comodità, ridenominiamo $R$ come $\prec$, dunque $f \prec g \iff f(x) \leq g(x), \forall x \in S$

### Riflessività :heavy_check_mark: 

$f \in A \implies f(x) \leq f(x), \forall x \in S \implies f \prec f$

### Antisimmetria :heavy_check_mark: 

$f \prec g \land g \prec f \implies f(x) \leq g(x) \land g(x) \leq f(x), \forall x \in S \implies f(x) = g(x), \forall x \in S \implies f = g$

### Transitività :heavy_check_mark: 

$f \prec g \land g \prec h \implies f(x) \leq g(x) \leq h(x), \forall x \in S \implies f \prec h$

### Totalità :no_entry_sign: 

Consideriamo:
- $S = \mathbb{Q}$
- $f \in A \mid f(x) = x^2$
- $g \in A \mid g(x) = x$

A questo punto si ha che
- $\frac{1}{2} \in S \implies f(\frac{1}{2}) = \frac{1}{4} \land g(\frac{1}{2}) = \frac{1}{2} \implies f(\frac{1}{2}) < g(\frac{1}{2})$
- $2 \in S \implies f(2) = 4 \land g(2) = 2 \implies g(2) < f(2)$

Quindi, possiamo concludere che $\exists 2 \in S \mid f(2) \not\leq g(2) \implies f \not\prec g$ e che $\exists \frac{1}{2} \in S \mid g(\frac{1}{2}) \not\leq f(\frac{1}{2}) \implies g \not\prec f$

dunque esistono almeno due elementi $f, g \in $A$ per cui $f \not\prec g \land g \not\prec f$, implicando che la relazione sia un ordine parziale e non totale
